alibi_detect/cd/sklearn/classifier.py

```diff
@@ -272,24 +273,36 @@ def _score(self, x: Union[np.ndarray, list]) \
             idx_oof_list.append(idx_te)
         probs_oof = np.concatenate(probs_oof_list, axis=0)
         idx_oof = np.concatenate(idx_oof_list, axis=0)
-        x_oof, y_oof = x[idx_oof], y[idx_oof]
+        y_oof = y[idx_oof]
         n_cur = y_oof.sum()
         n_ref = len(y_oof) - n_cur
         p_val, dist = self.test_probs(y_oof, probs_oof, n_ref, n_cur)
-        probs_sort = probs_oof[np.argsort(idx_oof)]
-        x_sort = x_oof[np.argsort(idx_oof)]
+        idx_sort = np.argsort(idx_oof)
+        probs_sort = probs_oof[idx_sort]
+        if isinstance(x, np.ndarray):
+            x_oof = x[idx_oof]
+            x_sort = x_oof[idx_sort]
+        else:
+            x_oof = [x[_] for _ in idx_oof]
+            x_sort = [x_oof[_] for _ in idx_sort]
         return p_val, dist, probs_sort[:n_ref, 1], probs_sort[n_ref:, 1], x_sort[:n_ref], x_sort[n_ref:]
 
     def _score_rf(self, x: Union[np.ndarray, list]) \
-            -> Tuple[float, float, np.ndarray, np.ndarray, np.ndarray, np.ndarray]:
+            -> Tuple[float, float, np.ndarray, np.ndarray, Union[np.ndarray, list], Union[np.ndarray, list]]:
         x_ref, x = self.preprocess(x)
         x, y = self.get_splits(x_ref, x, return_splits=False)  # type: ignore
         self.model.fit(x, y)
         # it is possible that some inputs do not have OOB scores. This is probably means
         # that too few trees were used to compute any reliable estimates.
         idx_oob = np.where(np.all(~np.isnan(self.model.oob_decision_function_), axis=1))[0]
         probs_oob = self.model.oob_decision_function_[idx_oob]
-        x_oob, y_oob = x[idx_oob], y[idx_oob]
+        y_oob = y[idx_oob]
+        if isinstance(x, np.ndarray):
+            x_oob = x[idx_oob]
+        elif isinstance(x, list):
+            x_oob = [x[_] for _ in idx_oob]
+        else:
+            raise TypeError(f'x needs to be of type np.ndarray or list and not {type(x)}.')
```

Comment:
Noting that this is different to the pattern in previous parts where we only have a single `if/else` and no raising of error otherwise.

Reply:
Indeed, because elsewhere a check has already been carried out during training for the TypeError so it can be dropped later. Here this isn't case.